includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -56,7 +56,11 @@ function get_columns() {
 					'title'           => __( 'Course', 'sensei-lms' ),
 					'last_activity'   => __( 'Last Activity', 'sensei-lms' ),
 					'completions'     => __( 'Completed', 'sensei-lms' ),
-					'average_percent' => __( 'Average Grade', 'sensei-lms' ),
+					'average_percent' => sprintf(
+						// translators: Placeholder value is the average grade of all courses.
+						__( 'Average Grade (%d%%)', 'sensei-lms' ),
+						esc_html( Sensei()->grading->get_courses_average_grade() )
```

Comment:
@donnapep  Hi Donna I know this is just draft, but wanted to point out something that I've discovered when doing totals for Students reports: 
```get_columns()``` function is being called 3 times. And only once to generate the table, other two times if from some VIP functions. 
The problem with this is that  get_courses_average_grade() function is also called 3 times. I tested it locally by print_r in the ```get_courses_average_grade()``` function, so you can see it here. On single page load it is called 3 times:  (Very savage but quick approach :D )
<img width="446" alt="image" src="https://user-images.githubusercontent.com/7208249/157429656-ba780555-80bd-40d4-ae76-05a67feeafac.png">

I solved that by moving the call to Sensei()->grading->get_courses_average_grade() in prepare data function. [Here](https://github.com/Automattic/sensei/pull/4865/files#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR214)

and storing the data to class properties:[Here]( https://github.com/Automattic/sensei/pull/4865/files#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR31
)

And using them like [this ](https://github.com/Automattic/sensei/pull/4865/files#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR102)

Reply:
@LavaGolem Thanks! I went down a rabbit hole trying to figure out why it's calling it multiple times. I *think* it may be related to [this issue](https://core.trac.wordpress.org/ticket/34564), but the solution was not obvious so I went with an approach similar to your's. The only difference is that I [added the variable to the Grading class](https://github.com/Automattic/sensei/pull/4904/commits/71e357afe814b36bc0d985f156c2436cd9a37a58#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR27) instead, which seems like a more appropriate place for it, and then [checked if the variable is set](https://github.com/Automattic/sensei/pull/4904/commits/71e357afe814b36bc0d985f156c2436cd9a37a58#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1221) before querying the data again.

WDYT?

Reply:
I think instead of using a variable in the Grading class, it would be better to fix the `get_columns` method so it's not called for each column. Here's a patch:

```patch
diff --git a/includes/class-sensei-analysis-overview-list-table.php b/includes/class-sensei-analysis-overview-list-table.php
index 98b63d828..18843bc38 100755
--- a/includes/class-sensei-analysis-overview-list-table.php
+++ b/includes/class-sensei-analysis-overview-list-table.php
@@ -50,6 +50,10 @@ class Sensei_Analysis_Overview_List_Table extends Sensei_List_Table {
         */
        function get_columns() {

+               if ( $this->columns ) {
+                       return $this->columns;
+               }
+
                switch ( $this->type ) {
                        case 'courses':
                                $columns = array(
@@ -89,7 +93,10 @@ class Sensei_Analysis_Overview_List_Table extends Sensei_List_Table {
                // Backwards compatible filter name, moving forward should have single filter name
                $columns = apply_filters( 'sensei_analysis_overview_' . $this->type . '_columns', $columns, $this );
                $columns = apply_filters( 'sensei_analysis_overview_columns', $columns, $this );
-               return $columns;
+
+               $this->columns = $columns;
+
+               return $this->columns;
        }

        /**
```

Reply:
Oh, you so smart! 🕺🏻 https://github.com/Automattic/sensei/pull/4904/commits/bfc499294d6f5614a33e7e5ffa7633bc43be6f22